apps/onnx/onnx_converter.cc

```diff
@@ -554,23 +554,22 @@ Node convert_metadata_node(
     result.outputs.resize(1);
     result.outputs[0].rep = func_for_node_output(node, 0);
 
-    std::vector<Halide::Expr> dims;
-    Halide::Expr num_elements = 1;
-    const std::vector<Halide::Expr> &input_shape = inputs[0].shape;
-    for (int i = 0; i < input_shape.size(); ++i) {
-        dims.push_back(input_shape[i]);
-        num_elements *= input_shape[i];
-    }
+    const std::vector<Halide::Expr>& input_shape = inputs[0].shape;
 
     if (node.op_type() == "Size") {
-        result.outputs[0].rep() = Halide::cast<int64_t>(num_elements);
+      Halide::Expr num_elements = 1;
+      for (int i = 0; i < input_shape.size(); ++i) {
+	num_elements *= input_shape[i];
+      }
+      result.outputs[0].rep() = Halide::cast<int64_t>(num_elements);
     } else {
-        std::vector<Halide::Var> var(1);
-        result.outputs[0].rep(var) = Halide::cast<int64_t>(0);
-        for (int i = 0; i < dims.size(); ++i) {
-            result.outputs[0].rep(i) = Halide::cast<int64_t>(dims[i]);
-        }
-        result.outputs[0].shape.push_back(static_cast<int>(input_shape.size()));
+      Halide::Var var;
+      Halide::Expr res = Halide::Expr(static_cast<int64_t>(0));
+      for (int i = 0; i < input_shape.size(); ++i) {
+	res = Halide::select(var == i, Halide::cast<int64_t>(input_shape[i]), res);
+      }
+      result.outputs[0].rep(var) = res;
```

Comment:
I don't know the semantics of the 'shape' op, so I can't comment on the correctness of the code here.